lib/Doctrine/ORM/Query/AST/Functions/AbsFunction.php

```diff
@@ -19,16 +19,14 @@ class AbsFunction extends FunctionNode
     /** @var SimpleArithmeticExpression */
     public $simpleArithmeticExpression;
```

Comment:
:thinking: public properties are not migrated it seems

Reply:
Which is best because that would not even pass the test suite:
```
1) Doctrine\Tests\ORM\Functional\QueryDqlFunctionTest::testFunctionAbs
TypeError: Cannot assign Doctrine\ORM\Query\AST\ArithmeticTerm to property Doctrine\ORM\Query\AST\Functions\AbsFunction::$simpleArithmeticExpression of type Doctrine\ORM\Query\AST\SimpleArithmeticExpression

/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/AST/Functions/AbsFunction.php:33
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:3531
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:3475
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:2282
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:1238
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:927
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:896
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:306
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:406
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query.php:244
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query.php:253
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:931
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:887
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:687
/home/greg/dev/doctrine-orm-split/major/tests/Doctrine/Tests/ORM/Functional/QueryDqlFunctionTest.php:72

2) Doctrine\Tests\ORM\Functional\Ticket\GH7941Test::typesShouldBeConvertedForDQLFunctions
TypeError: Cannot assign Doctrine\ORM\Query\AST\PathExpression to property Doctrine\ORM\Query\AST\Functions\AbsFunction::$simpleArithmeticExpression of type Doctrine\ORM\Query\AST\SimpleArithmeticExpression

/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/AST/Functions/AbsFunction.php:33
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:3531
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:3475
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:2282
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:1233
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:927
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:896
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:306
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query/Parser.php:406
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query.php:244
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/Query.php:253
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:931
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:887
/home/greg/dev/doctrine-orm-split/major/lib/Doctrine/ORM/AbstractQuery.php:687
/home/greg/dev/doctrine-orm-split/major/tests/Doctrine/Tests/ORM/Functional/Ticket/GH7941Test.php:75
```

Reply:
I migrated all properties having `Node` as a type hint, and left most other properties alone. I think this will deserve a separate PR if we decide to do something about it.

Reply:
That was a good call, I think. Many property types in the `AST` namespace are inaccurate. Migrating the straightforward ones first is probably the best we can do.